src/inference/src/dev/core_impl.cpp

```diff
@@ -936,9 +954,21 @@ ov::SoPtr<ov::ICompiledModel> ov::CoreImpl::compile_model(const std::string& mod
         cacheContent.blobId =
             ov::ModelCache::compute_hash(model_str, weights, create_compile_config(plugin, parsed._config));
         std::unique_ptr<CacheGuardEntry> lock = cacheGuard.get_hash_lock(cacheContent.blobId);
+        
```

Comment:
The parameter `const ov::Tensor& weights` may contains original weight so reading model from string can be not required. 
Configuration may have also have got hint to weight path which can be used, will better if device can decide if model read from stream is required or not in weightless mode, as this is GPU specific only.